src/meta/src/manager/hash_mapping.rs

```diff
@@ -91,6 +93,25 @@ impl HashMappingManager {
             .map(|info| info.vnode_mapping.clone())
     }
 
+    pub fn check_sst_deprecated(&self, vnode_bitmaps: &[VNodeBitmap]) -> bool {
+        let core = self.core.lock();
+        for vnode_bitmap in vnode_bitmaps {
+            if let Some(fragment_id) = core
+                .state_table_fragment_mapping
+                .get(&vnode_bitmap.get_table_id())
+            {
+                if let Some(HashMappingInfo { map_hsh, .. }) =
+                    core.hash_mapping_infos.get(fragment_id)
+                {
+                    if *map_hsh != vnode_bitmap.get_map_hsh() {
```

Comment:
Does it means that there the hash-distribution for compaction has changed so that we need to compact this file?